contracts/staking/PoSValidatorManager.sol

```diff
@@ -264,15 +278,16 @@ abstract contract PoSValidatorManager is IPoSValidatorManager, ValidatorManager
         PoSValidatorManagerStorage storage $ = _getPoSValidatorManagerStorage();
         bytes32 validationID = $._delegatorStakes[delegationID].validationID;
 
+        Delegator memory delegator = $._delegatorStakes[delegationID];
+
         uint64 uptime;
         if (includeUptimeProof) {
             uptime = _getUptime(validationID, messageIndex);
         }
-
-        // TODO: Calculate the delegator's reward, but do not unlock it
+        $._pendingDelegatorRewards[delegationID] =
+            $._rewardCalculator.calculateReward(delegator.weight, uptime, 0, 0);
```

Comment:
We definitely shouldn't be passing the validator's uptime into the reward calculator. Currently if a delegator staked for one second, they would get rewards for the entire validator's uptime.